hibernate6-jpa/src/main/java/org/hibernate/metamodel/internal/EntityRepresentationStrategyPojoStandard.java

```diff
@@ -0,0 +1,441 @@
+
+	private final String identifierPropertyName;
+	private final PropertyAccess identifierPropertyAccess;
+	private final Map<String, PropertyAccess> propertyAccessMap = new ConcurrentHashMap<>();
+	private final EmbeddableRepresentationStrategyPojo mapsIdRepresentationStrategy;
+
+	public EntityRepresentationStrategyPojoStandard(
+			PersistentClass bootDescriptor,
+			EntityPersister runtimeDescriptor,
+			RuntimeModelCreationContext creationContext) {
+		final SessionFactoryImplementor sessionFactory = creationContext.getSessionFactory();
+		final JavaTypeRegistry jtdRegistry = creationContext.getTypeConfiguration()
+				.getJavaTypeRegistry();
+
+		final Class<?> mappedJavaType = bootDescriptor.getMappedClass();
+		this.mappedJtd = jtdRegistry.resolveEntityTypeDescriptor( mappedJavaType );
+
+		final Class<?> proxyJavaType = bootDescriptor.getProxyInterface();
+		if ( proxyJavaType != null ) {
+			this.proxyJtd = jtdRegistry.getDescriptor( proxyJavaType );
+		}
+		else {
+			this.proxyJtd = null;
+		}
+
+		this.lifecycleImplementor = Lifecycle.class.isAssignableFrom( mappedJavaType );
+		this.isBytecodeEnhanced = isPersistentAttributeInterceptableType( mappedJavaType );
+
+		final Property identifierProperty = bootDescriptor.getIdentifierProperty();
+		if ( identifierProperty == null ) {
+			identifierPropertyName = null;
+			identifierPropertyAccess = null;
+
+			final KeyValue bootDescriptorIdentifier = bootDescriptor.getIdentifier();
+
+			if ( bootDescriptorIdentifier instanceof Component ) {
+				if ( bootDescriptor.getIdentifierMapper() != null ) {
+					mapsIdRepresentationStrategy = new EmbeddableRepresentationStrategyPojo(
+							bootDescriptor.getIdentifierMapper(),
+							() -> ( ( CompositeTypeImplementor) bootDescriptor.getIdentifierMapper().getType() )
+									.getMappingModelPart().getEmbeddableTypeDescriptor(),
+							// we currently do not support custom instantiators for identifiers
+							null,
+							null,
+							creationContext
+					);
+				}
+				else if ( bootDescriptorIdentifier != null ) {
+					mapsIdRepresentationStrategy = new EmbeddableRepresentationStrategyPojo(
+							(Component) bootDescriptorIdentifier,
+							() -> ( ( CompositeTypeImplementor) bootDescriptor.getIdentifierMapper().getType() )
+									.getMappingModelPart().getEmbeddableTypeDescriptor(),
+							// we currently do not support custom instantiators for identifiers
+							null,
+							null,
+							creationContext
+					);
+				}
+				else {
+					mapsIdRepresentationStrategy = null;
+				}
+			}
+			else {
+				mapsIdRepresentationStrategy = null;
+			}
+		}
+		else {
+			mapsIdRepresentationStrategy = null;
+			identifierPropertyName = identifierProperty.getName();
+			identifierPropertyAccess = makePropertyAccess( identifierProperty );
+		}
+
+		final BytecodeProvider bytecodeProvider = creationContext.getBootstrapContext().getServiceRegistry().getService(BytecodeProvider.class);
```

Comment:
This is class copied from Hibernate and changed this line in order to use bytecode provider that we registered. Hibernate 6 uses `Environment.getBytecodeProvider();` and seems like there is no way to use custom. There is JIRA issue for this and I asked also in Hibernate ORM discussions about it.

Reply:
Removed, no longer needed since we don't support compile time proxy in hibernate6